static/app/components/core/badge/alertBadge.tsx

```diff
@@ -41,50 +42,88 @@ interface AlertBadgeProps {
-  } else if (status === IncidentStatus.WARNING) {
-    statusText = t('Warning');
-    Icon = IconExclamation;
-    color = 'warningText';
-  }
+export function AlertBadge(props: AlertBadgeProps) {
+  const theme = useTheme();
+  const {text, icon: Icon} = getDiamondTheme(
+    props.status,
+    props.isIssue,
+    props.isDisabled,
+    theme
+  );
 
   return (
-    <Wrapper data-test-id="alert-badge">
-      <DiamondStatus
-        icon={Icon}
-        color={color}
-        aria-label={!withText ? statusText : undefined}
-      />
-      {withText && <div>{statusText}</div>}
-    </Wrapper>
+    <PaddedContainer data-test-id="alert-badge" align="center" gap={space(1.5)}>
+      <DiamondBackground
+        {...props}
+        role="presentation"
+        aria-label={!props.withText ? text : undefined}
+      >
+        <Icon width={13} height={13} />
+      </DiamondBackground>
+      {props.withText && <div>{text}</div>}
+    </PaddedContainer>
   );
 }
 
-/**
- * The size of the issue icon needs to be marginally adjusted to fit into the diamond well
- */
-const SizedIconIssue = styled(IconIssues)`
-  width: 13px;
-`;
+function getDiamondTheme(
+  status: AlertBadgeProps['status'],
+  isIssue: AlertBadgeProps['isIssue'],
+  isDisabled: AlertBadgeProps['isDisabled'],
+  theme: Theme
+): {
+  backgroundColor: string;
+  icon: React.ComponentType<SVGIconProps>;
+  text: string;
+} {
+  if (isDisabled) {
+    return {text: t('Disabled'), backgroundColor: theme.disabled, icon: IconPause};
+  }
+  if (isIssue) {
+    return {
+      text: t('Issue'),
+      backgroundColor: theme.subText,
+      // @TODO(jonasbadalic): why does the issues icon height need to be adjusted?
+      icon: (props: SVGIconProps) => <IconIssues width={13} height={13} {...props} />,
+    };
+  }
+  if (status === IncidentStatus.CRITICAL) {
+    return {text: t('Critical'), backgroundColor: theme.errorText, icon: IconFire};
+  }
+  if (status === IncidentStatus.WARNING) {
+    return {
+      text: t('Warning'),
+      backgroundColor: theme.warningText,
+      icon: IconExclamation,
+    };
+  }
+  return {text: t('Resolved'), backgroundColor: theme.successText, icon: IconCheckmark};
+}
 
-const Wrapper = styled('div')`
-  display: flex;
-  align-items: center;
-  gap: ${space(1)};
+const PaddedContainer = styled(Flex)`
+  /* @TODO(jonasbadalic): This used to be sized by the oversized icon inside it */
+  padding: 5px 4px;
 `;
+
+const DiamondBackground = withChonk(
+  styled('div')<AlertBadgeProps>`
+    background-color: ${p =>
+      getDiamondTheme(p.status, p.isIssue, p.isDisabled, p.theme).backgroundColor};
+    width: 26px;
+    height: 26px;
+
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    border-radius: 2px;
+    transform: rotate(45deg);
+
+    > svg {
+      width: 13px;
+      height: 13px;
+      transform: rotate(-45deg);
+      color: ${p => p.theme.white};
+    }
+  `,
+  ChonkAlertBadgeDiamondBackground,
+  p => p
```

Comment:
thought: if we have this more often, maybe we should make it a default?

Reply:
Yeah, I think that makes sense.